cert/file.go

```diff
@@ -126,25 +128,31 @@ func (f *File) parse() (err error) {
 	return nil
 }
 
-// Set ensures the file has the right owner/group and mode.
-func (f *File) setPermissions() error {
+// CheckPermissions checks that the permissions on disk match what we intend.  If the
+// file doesn't exist, then the raw error for stat is returned.
+func (f *File) CheckPermissions() error {
 	st, err := os.Stat(f.Path)
 	if err != nil {
 		return err
 	}
 
-	err = os.Chown(f.Path, f.uid, f.gid)
-	if err != nil {
-		return err
+	if st.Mode() != f.mode {
+		return fmt.Errorf("mode has changed from %s to %s", f.mode, st.Mode())
 	}
 
-	if st.Mode() != f.mode {
-		err = os.Chmod(f.Path, f.mode)
-		if err != nil {
-			return err
-		}
+	unixStat, ok := st.Sys().(*syscall.Stat_t)
+	if !ok {
+		// yes, this is noisy on windows.  Don't think we have windows users however...
```

Comment:
We probably have BSD users though. I *think* this should work fine on BSD, but that might be worth a check.

Reply:
That conversion should work for all unixes; if it doesn't, the warning is fairly clear and they'll turn up with a patch or a ticket.

That's about the best I can test for this at this juncture.